packages/assets-controllers/src/NftDetectionController.ts

```diff
@@ -588,6 +611,72 @@ export class NftDetectionController extends BaseController<
               ? elm.blockaidResult?.result_type === BlockaidResultType.Benign
               : true),
         );
+        // Retrieve collections from apiNfts
+        const collections = apiNfts.reduce<string[]>((acc, currValue) => {
+          if (!acc.includes(currValue.token.contract)) {
+            acc.push(currValue.token.contract);
+          }
+          return acc;
+        }, []);
+
+        // Call API to retrive collections infos
+        // The api accept a max of 20 contracts
+        const collectionResponse: GetCollectionsResponse =
+          await reduceInBatchesSerially({
+            values: collections,
+            batchSize: MAX_GET_COLLECTION_BATCH_SIZE,
+            eachBatch: async (allResponses, batch) => {
+              const params = new URLSearchParams(
+                batch.map((s) => ['contract', s]),
+              );
+              params.append('chainId', '1'); // Adding chainId 1 because we are only detecting for mainnet
+              const collectionResponseForBatch = await fetchWithErrorHandling({
+                url: `${
+                  NFT_API_BASE_URL as string
+                }/collections?${params.toString()}`,
+                options: {
+                  headers: {
+                    Version: NFT_API_VERSION,
+                  },
+                },
+                timeout: NFT_API_TIMEOUT,
+              });
+
+              return {
+                ...allResponses,
+                ...collectionResponseForBatch,
+              };
+            },
+            initialResult: {},
+          });
+
+        // Add collections response fields to  newnfts
```

Comment:
Something I didn't realize is that we already have a large collection sub-field under each NFT, since it comes from reservoir that way.  Which is unnecessary duplicated state when you have a bunch of NFTs in the same collection.  Since it should be the same for all of them.

We do have an existing field `allNftContracts` where we could choose to put this sort of stuff in the future.  what do you think?

Reply:
I agree that when you have a 100 NFTs of the same collection, you will have the same collection field in each NFT object.I think that is one way of refactoring the logic in the future! 

Although the fact that we have collection under each NFT is useful for cases like artblocks contracts; 
but we can update the type of the `allNftContracts` to maybe also include a `tokenId` so on the client side we are able to fetch the correct collection information from `allNftContracts`.

An example of what i was refurring to is this user address: `0x484585fe5B976C2AFAC0f2c912447059EF41bf78`
when you try to check NFTs held and filter with this contract address: `0x99a9B7c1116f9ceEB1652de04d5969CcE509B069` (artblocks)
The returned NFTs would include this [NFT](https://opensea.io/assets/ethereum/0x99a9b7c1116f9ceeb1652de04d5969cce509b069/486000043) and this [NFT](https://opensea.io/assets/ethereum/0x99a9b7c1116f9ceeb1652de04d5969cce509b069/489000124)